src/realm/array.hpp

```diff
@@ -1991,33 +1795,25 @@ inline bool Array::compare_equality(int64_t value, size_t start, size_t end, siz
 // takes as most as possible to get best performance.
 
 // This is the one installed into the m_vtable->finder slots.
-template <class cond, Action action, size_t bitwidth>
-bool Array::find(int64_t value, size_t start, size_t end, size_t baseindex, QueryState<int64_t>* state) const
+template <class cond, size_t bitwidth>
+bool Array::find_vtable(int64_t value, size_t start, size_t end, size_t baseindex, QueryStateBase* state) const
 {
-    return find<cond, action, bitwidth>(value, start, end, baseindex, state, CallbackDummy());
+    return find_optimized<cond, bitwidth>(value, start, end, baseindex, state, nullptr);
 }
 
-template <class cond, Action action, class Callback>
-bool Array::find(int64_t value, size_t start, size_t end, size_t baseindex, QueryState<int64_t>* state,
-                 Callback callback, bool nullable_array, bool find_null) const
+template <class cond, class Callback>
+bool Array::find(int64_t value, size_t start, size_t end, size_t baseindex, QueryStateBase* state, Callback callback,
```

Comment:
Have you measured if it makes a difference to pass `Callback` by reference instead of by value? There could be some code bloaty from copy constructors.

Reply:
No I have not. We can do that as a follow up action.